`go_staticcheck_lint_package` really should default to 1 instead of 0

this is a minimal example:

```
$ cat go.mod
module foo

go 1.16
$ cat type.go
package foo

type Foo int
$ cat code.go
package foo

func Bar() {
        var _ Foo
}
```

`staticcheck .` would report no error, which is expected (this is what `let g:ale_go_staticcheck_lint_package = 1` does), but `staticcheck code.go` would complain (this is what `let g:ale_go_staticcheck_lint_package = 0` does, which is the current default value):

```
$ staticcheck code.go 
code.go:4:8: undeclared name: Foo (compile)
```

I'm not sure in which scenario `go_staticcheck_lint_package = 0` is preferred.

A PR will follow shortly.

--- a/doc/ale-go.txt
+++ b/doc/ale-go.txt
@@ -354,7 +354,7 @@ g:ale_go_staticcheck_options                     *g:ale_go_staticcheck_options*
 g:ale_go_staticcheck_lint_package           *g:ale_go_staticcheck_lint_package*
                                             *b:ale_go_staticcheck_lint_package*
   Type: |Number|
-  Default: `0`
+  Default: `1`
 
   When set to `1`, the whole Go package will be checked instead of only the
   current file.

